frontend/src/components/tools/shared/FileStatusIndicator.tsx

```diff
@@ -12,13 +15,39 @@ const FileStatusIndicator = ({
   placeholder,
 }: FileStatusIndicatorProps) => {
   const { t } = useTranslation();
-  const defaultPlaceholder = placeholder || t("files.placeholder", "Select a PDF file in the main view to get started");
-  
-  // Only show content when no files are selected
+  const { openFilesModal } = useFilesModalContext();
+  const { files: workbenchFiles } = useAllFiles();
+
+  // Check if there are no files in the workbench
+  if (workbenchFiles.length === 0) {
+    return (
+      <Text size="sm" c="dimmed">
+        {t("files.noFiles", "No files uploaded. ")}{" "}
+        <Anchor
+          size="sm"
+          onClick={openFilesModal}
+          style={{ cursor: 'pointer', display: 'inline-flex', alignItems: 'center', gap: '4px' }}
+        >
+          <FolderIcon style={{ fontSize: '14px' }} />
+          {t("files.addFiles", "Add files")}
+        </Anchor>
+      </Text>
+    );
+  }
+
+  // Show selection status when there are files in workbench
   if (selectedFiles.length === 0) {
     return (
       <Text size="sm" c="dimmed">
-        {defaultPlaceholder}
+        {t("files.selectFromWorkbench", "Select files from the workbench or ")}{" "}
```

Comment:
```suggestion
        {t("files.selectFromWorkbench", "Select files from the workbench or ") + " "}
```

* This isn't in the translations file as far as I can see
* I don't really like the wording of this, but I can see exactly what you're going for and it's not _wrong_. It's just that it reads like a sentence that doesn't end followed by a button on the next line to me. I'm not going to push hard for it to be changed because I don't have any suggestions of other wordings. The `Add Files` one does read properly for me now at least 🙂